### PR TITLE
Add Codex

### PR DESCRIPTION
**Pull request by cycool29**
https://github.com/jcv8000/Codex

Great note-taking tool for programmers.

### CODE REVIEW

File: apps/Codex/install [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+DIRECTORY="$(dirname "$(dirname "$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )")")"
+
+function error {
+  echo -e "\\e[91m$1\\e[39m"
+  exit 1
+}
+
+# Install Node.js first
+"${DIRECTORY}/manage" install-if-not-installed "Node.js" "$(dirname "$0")" || error "Failed to install Node.js"
+"${DIRECTORY}/pkg-install" "git" "$(dirname "$0")" || exit 1
```
Comment (Itai-Nelken) [inline apps/Codex/install:12]: No need to install git as it's installed when pi-apps is installed.

File: apps/Codex/install [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+
+# Install Codex
+( npm install || sudo npm install ) 2>&1 | grep -v "deprecated popper.js"
+
+# Create run-codex script
+echo "export NVM_DIR='$HOME/.nvm'
+[ -s '$NVM_DIR/nvm.sh' ] && \. '$NVM_DIR/nvm.sh'  # This loads nvm
+[ -s '$NVM_DIR/bash_completion' ] && \. '$NVM_DIR/bash_completion'  # This loads nvm bash_completion
```
Comment (Itai-Nelken) [inline apps/Codex/install:34]: No need for the nvm bash completion here.

File: apps/Codex/install [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+rm -rf Codex || sudo rm -rf Codex
+# Clone Codex repo
+git clone https://github.com/jcv8000/Codex
+cd ./Codex
+CODEX_DIR=$(pwd)
+
+# One time use, since `source ~/.bashrc` not working
+export NVM_DIR="$HOME/.nvm"
+[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
```
Comment (Itai-Nelken) [inline apps/Codex/install:26]: There is no need for the nvm bash completion here.

**Review by Itai-Nelken — CHANGES_REQUESTED** [review]
Looks good. Is this tested?

File: apps/Codex/install (hunk 3 above)
Comment (cycool29) [reply apps/Codex/install:26]: Tested before, the `npm install` command won't work without the bash completion.

File: apps/Codex/install (hunk 2 above)
Comment (cycool29) [reply apps/Codex/install:34]: Same, I don't know why `source $HOME/.bashrc` not working, so I will just source the `nvm.sh` again before start the program.

File: apps/Codex/install (hunk 1 above)
Comment (cycool29) [reply apps/Codex/install:12]: Thanks, will remove.

**Comment by cycool29** [comment]
@Itai-Nelken I've removed the `pkg install git` and added some comments in the install script.

**Comment by cycool29** [comment]
@Botspot What do you think?

**Comment by Botspot** [comment]
Tested and working. Scripts look good.

**Pull request merged**